File: frontend/src/types/import-export.ts
export interface ExportBook {
  title: string
  author: string
  description: string | null
  thumbnail: string | null
  pages: number | null
  published_date: string | null
  google_books_id: string | null
  genres: string[]
  status: string
  current_page: number | null
  rating: number | null
  started_at: string | null
  finished_at: string | null
  notes: { content: string; page_number: number | null }[]
}

export interface ExportPayload {
  version: number
  exported_at: string | null
  books: ExportBook[]
}

export interface ImportPreviewBook {
  row: number
  title: string
  author: string
  exists: boolean
  reason: string
}

export interface ImportPreviewResponse {
  total: number
  books: ImportPreviewBook[]
}

File: backend/app/schemas/import_export.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class ExportBookOut(BaseModel):
    title: str
    author: str
    description: Optional[str] = None
    thumbnail: Optional[str] = None
    pages: Optional[int] = None
    published_date: Optional[str] = None
    google_books_id: Optional[str] = None
    genres: list[str] = []
    status: str
    current_page: Optional[int] = None
    rating: Optional[float] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    notes: list["ExportNoteOut"] = []


class ExportNoteOut(BaseModel):
    content: str
    page_number: Optional[int] = None


class ImportPreviewBook(BaseModel):
    row: int
    title: str
    author: str
    exists: bool
    reason: str


class ImportPreviewResponse(BaseModel):
    total: int
    books: list[ImportPreviewBook]
